### PR TITLE
Learn different ways of writing in file

### PR DESCRIPTION
**Pull request by TimurKM**
Please, review

### CODE REVIEW

File: TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#include <iostream>
+#include <fstream>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp:2]: Please check the headers.

File: TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include <iostream>
+#include <fstream> 
+#include <Windows.h>
+#include <string>
+using namespace std;
+
+int main()
+{
+	ofstream file;
+	file.open("D:\\������������\\TestFile.txt");
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp:10]: Use std::cin to receive input from user.

File: TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	file << "Hello, world!";
+	file.close();
+
+	ofstream file("D:\\������������\\TestFile.txt", ios::app);
+	file << "Hello, new world!" << endl;
+	file.close();
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp:16]: Check return values.

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <fstream>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp:2]: Check headers.

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	if (hFile == INVALID_HANDLE_VALUE)
+	{
+		// Failed to open/create file
+		return 2;
+	}
+
+	// Write data to the file
+	std::string strText = "Hello World!"; // For C use LPSTR (char*) or LPWSTR (wchar_t*)
+	DWORD bytesWritten;
+	WriteFile(
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp:28]: Check the return value.

**Review by dimagerasimov — CHANGES_REQUESTED** [review]

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+#include <fstream> 
+#include <Windows.h>
+#include <string>
+
+using namespace std;
+
+int main()
+{
+	FILE* f;
+	f = fopen("D:\\������������\\Testfile3.txt", "w");
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp:11]: scanf to get a filename from user.

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#include <iostream>
+#include <fstream>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp:2]: Check headers.

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp (hunk 6 above)
Comment (dimagerasimov) [reply TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp:11]: Check return values for fopen, fprintf, fclose.

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp [hunk 8]
```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <fstream> 
+#include <string>
+
+int main()
+{
+	// Open a handle to the file
+	HANDLE hFile = CreateFile(
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp:8]: Use std::getline to get a filename from user.

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	DWORD bytesWritten;
+	WriteFile(
+		hFile,            // Handle to the file
+		strText.c_str(),  // Buffer to write
+		strText.size(),   // Buffer size
+		&bytesWritten,    // Bytes written
+		nullptr);         // Overlapped
+
+	 // Close the handle once we don't need it.
+	CloseHandle(hFile);
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp:34]: Check the return value.

File: TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp [hunk 10]
```diff
@@ -0,0 +1,14 @@
+#include <iostream>
+#include <fstream> 
+
+int main()
+{
+	std::ofstream file;
+	char name[] = "";
+	std::cin >> name;
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay1/TaskWithWritingInFileWay1.cpp:8]: You overflow the array name. What happens if user input is more than sizeof(name)=1? Please declare name as std::string in this example.

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp [hunk 11]
```diff
@@ -0,0 +1,46 @@
+#include <windows.h>
+#include <string>
+#include <iostream>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp:3]: It seems nothing from the iostream is used. Please remove the include.

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+#include <errhandlingapi.h>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp:4]: The errhandlingapi.h must be a part of Windows.h (which you include in 1st line). You can remove and check the code can compile

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp [hunk 13]
```diff
@@ -0,0 +1,25 @@
+#include <windows.h>
+#include <fstream>
+#include <string>
+#include <iostream>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp:4]: iostream not used.

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp [hunk 14]
```diff
@@ -0,0 +1,25 @@
+#include <windows.h>
+#include <fstream>
+#include <string>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp:3]: string not used

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp [hunk 15]
```diff
@@ -0,0 +1,25 @@
+#include <windows.h>
```
Comment (dimagerasimov) [inline TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp:1]: windows.h not used

**Review by dimagerasimov — CHANGES_REQUESTED** [review]

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp (hunk 11 above)
Comment (TimurKM) [reply TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp]: Removed

File: TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp (hunk 12 above)
Comment (TimurKM) [reply TaskWithWritingInFile/TaskWithWritingInFileWay2/TaskWithWritingInFileWay2.cpp]: Removed

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp (hunk 13 above)
Comment (TimurKM) [reply TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp]: Removed

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp (hunk 14 above)
Comment (TimurKM) [reply TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp]: Removed

File: TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp (hunk 15 above)
Comment (TimurKM) [reply TaskWithWritingInFile/TaskWithWritingInFileWay3/TaskWithWritingInFile.cpp]: Removed

**Review by dimagerasimov — APPROVED** [review]

**Pull request merged**